### PR TITLE
[Security] Reference the new request_matcher option

### PR DESCRIPTION
**Pull request by TristanPouliquen**
Linked to https://github.com/symfony/symfony/pull/44670

### CODE REVIEW

**Comment by derrabus** [comment]
Please target 6.1 here as well.

**Comment by derrabus** [comment]
I think, the documentation should show an actual code example of a configured request matcher instead of just mentioning that it is somehow possible to provide your own matcher.

**Comment by TristanPouliquen** [comment]
I tried adding one copying from other similar configuration code blocks, is it ok/enough?

**Comment by derrabus** [comment]
We usually don't write whole documentation blocks inside a `versionadded` block. Just document the feature as if it has always existed and add a small `versionadded` block blow that new piece of documentation where you tell the reader that this feature has actually been added in 6.1. Once we work on the documentation for Symfony 7, a maintainer can simply drop that `versionadded` block without touching the rest of your documentation.

File: security.rst [hunk 1]
```diff
@@ -2117,7 +2117,7 @@ Prepending the path with ``^`` means that only URLs *beginning* with the
 pattern are matched. For example, a path of ``/admin`` (without the ``^``)
 would match ``/admin/foo`` but would also match URLs like ``/foo/admin``.
 
-Each ``access_control`` can also match on IP address, hostname and HTTP methods.
+Each ``access_control`` can also match on IP address, hostname and HTTP methods or be directly configured by passing a service implementing `RequestMatcherInterface` through the `request_matcher` option.
```
Comment (derrabus) [inline security.rst]: I would split this into two sentences to make this piece of information easier to digest.

Comment (TristanPouliquen) [reply security.rst]: done :)

**Comment by TristanPouliquen** [comment]
@derrabus Is it ok for you to merge now? :)

**Review by derrabus — APPROVED** [review]
LGTM, but I'm powerless on this repository. Someone from the docs team will need to review and merge your PR. 🙃

**Comment by TristanPouliquen** [comment]
> LGTM, but I'm powerless on this repository. Someone from the docs team will need to review and merge your PR. upside_down_face

Ok, sorry for the ping then, who should I notify to see it merged? :)

File: security.rst [hunk 2]
```diff
@@ -2119,6 +2119,12 @@ would match ``/admin/foo`` but would also match URLs like ``/foo/admin``.
 
 Each ``access_control`` can also match on IP address, hostname and HTTP methods.
 It can also be used to redirect a user to the ``https`` version of a URL pattern.
+
+.. versionadded:: 6.1
+
+    Since Symfony 6.1, an access control rule can also be directly configured by passing a service
+    implementing `RequestMatcherInterface` through the `request_matcher` option.
```
Comment (nicolas-grekas) [inline security.rst:2126]: backticks should be doubled

**Comment by javiereguiluz** [comment]
Tristan, sorry it took us so long to merge your contribution 🙏  ... and congrats on your first Symfony Docs contribution 🎉

**Pull request merged**